File: app/src/main/java/com/tunjid/androidbootstrap/viewholders/ScanViewHolder.java
package com.tunjid.androidbootstrap.viewholders;

import android.view.View;
import android.widget.TextView;

import com.tunjid.androidbootstrap.R;
import com.tunjid.androidbootstrap.adapters.ScanAdapter;
import com.tunjid.androidbootstrap.communications.bluetooth.ScanResultCompat;
import com.tunjid.androidbootstrap.recyclerview.InteractiveViewHolder;

public class ScanViewHolder extends InteractiveViewHolder<ScanAdapter.ScanAdapterListener>
         implements
         View.OnClickListener {

     private TextView deviceName;
     private TextView deviceAddress;

     private ScanResultCompat result;

    public ScanViewHolder(View itemView, ScanAdapter.ScanAdapterListener scanAdapterListener) {
         super(itemView, scanAdapterListener);

         deviceAddress = itemView.findViewById(R.id.device_address);
         deviceName = itemView.findViewById(R.id.device_name);
         itemView.setOnClickListener(this);
     }

    public void bind(ScanResultCompat result) {
         this.result = result;
         if (result.getScanRecord() != null) {
             deviceName.setText(result.getScanRecord().getDeviceName());
             deviceAddress.setText(result.getDevice().getAddress());
         }
     }

     @Override
     public void onClick(View v) {
         switch ((v.getId())) {
             case R.id.row_parent:
                 adapterListener.onBluetoothDeviceClicked(result.getDevice());
                 break;
         }
     }
 }

File: app/src/main/java/com/tunjid/androidbootstrap/viewholders/ScanViewHolder.kt
package com.tunjid.androidbootstrap.viewholders

import android.view.View
import android.widget.TextView

import com.tunjid.androidbootstrap.R
import com.tunjid.androidbootstrap.adapters.ScanAdapter
import com.tunjid.androidbootstrap.communications.bluetooth.ScanResultCompat
import com.tunjid.androidbootstrap.recyclerview.InteractiveViewHolder

class ScanViewHolder(itemView: View, scanAdapterListener: ScanAdapter.ScanAdapterListener)
    : InteractiveViewHolder<ScanAdapter.ScanAdapterListener>(itemView, scanAdapterListener), View.OnClickListener {

    private val deviceName: TextView = itemView.findViewById(R.id.device_name)
    private val deviceAddress: TextView = itemView.findViewById(R.id.device_address)

    private lateinit var result: ScanResultCompat

    init {
        itemView.setOnClickListener(this)
    }

    fun bind(result: ScanResultCompat) {
        this.result = result
        if (result.scanRecord != null) {
            deviceName.text = result.scanRecord!!.deviceName
            deviceAddress.text = result.device.address
        }
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.row_parent -> delegate.onBluetoothDeviceClicked(result.device)
        }
    }
}
